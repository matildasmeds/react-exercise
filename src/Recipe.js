import React, { useState } from 'react';
import Requests from './Requests.js';
import { Button } from './BasicComponents.js';

function Recipe(props) {
  const [isDeleted, setIsDeleted] = useState(false);

  const ingredients_html = props.ingredients.map((ingredient, index) =>
      <Ingredient name={ingredient.name} key={index} />
    );

  function deleteRecipe() {
    Requests.deleteRecipe(props.id).then(response => {
      if (response.status === 204) {
        setIsDeleted(true);
      }
    }, error => {
      console.log(error);
    });
  }

  const recipeHtml = <div className="w-100 mv2">
    <h3 className="f4 fwl baskerville mt0 lh-title">{props.name}</h3>
    <p>{props.description}</p>
    <div>{ingredients_html}</div>
    <div className="mt3">
      <a href={'/recipes/' + props.id + '/edit' }><Button label='Edit' /></a>
      <Button label='Delete' callBack={deleteRecipe} bgcolor='#ffaaa5' />
    </div>
  </div>

  const deletedHtml = <div className="w-100 mv2">
    <i>Deleted recipe...</i>
  </div>

  return <section className="bb b--black-10">
    { isDeleted ? deletedHtml : recipeHtml }
  </section>;
}

function Ingredient(props) {
  return <div>{props.name}</div>;
}

export default Recipe;
