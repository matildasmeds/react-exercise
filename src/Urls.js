const Urls = {
  recipes: {
    list: 'http://localhost:8000/api/recipe',
    create: ''
  }
};

export default Urls;
