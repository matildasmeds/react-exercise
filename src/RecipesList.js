import React, { useState, useEffect } from 'react';
import Recipe from './Recipe.js';
import { H1 } from './BasicComponents.js';
import axios from 'axios';
import Urls from './Urls.js'

function RecipesList() {
  const [recipes, setRecipes] = useState([]);

  useEffect(() => {
    axios(Urls.recipes.list).then(result => {
      setRecipes(result.data);
    });
  }, []);

  console.log(recipes);

  return <div className="pa4 black-80">
      <H1 label='All recipes' />
      { recipes && recipes.map(recipe =>
        <div key={recipe.id}>
          <Recipe
            name={recipe.name}
            description={recipe.description}
            ingredients={recipe.ingredients}
          />
         </div>)}
    </div>;
};

export default RecipesList;
