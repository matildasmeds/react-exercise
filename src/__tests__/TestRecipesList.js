import '@testing-library/jest-dom/extend-expect';

import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import RecipesList from '../RecipesList.js';
import Recipe from '../Recipe.js';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import Urls from '../Urls.js';

// Mock call to backend that returns some items
// Snapshot?
test('renders the list correctly', () => {
  const responseData = [{
    name: 'My favorite recipe',
    description: 'Easy, healthy, cheap, delicious',
    ingredients: [{
      name: 'Secret ingredient'
    }]
  }];
  const mock = new MockAdapter(axios);
  mock.onGet(Urls.recipes.list).reply(200, responseData);
  const recipeList = renderer.create(<RecipesList />);
  const recipesListTree = recipeList.toJSON();
  console.log(recipesListTree);

  expect(recipesListTree).toMatchSnapshot();
});
