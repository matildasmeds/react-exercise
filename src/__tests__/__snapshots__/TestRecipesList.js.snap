// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the list correctly 1`] = `
<div
  className="pa4 black-80"
>
  <h1
    className="baskerville fw1 ph3 ph0-l bb b--black-10 mb2 pb1"
  >
    All recipes
  </h1>
</div>
`;
